Source/WebKit/Platform/IPC/FormDataReference.h

```diff
@@ -40,53 +40,37 @@ class FormDataReference {
     {
     }
 
-    RefPtr<WebCore::FormData> takeData() { return WTFMove(m_data); }
-
-    void encode(Encoder& encoder) const
-    {
-        encoder << !!m_data;
-        if (!m_data)
-            return;
-
-        encoder << *m_data;
+    FormDataReference(RefPtr<WebCore::FormData>&&, Vector<WebKit::SandboxExtension::Handle>&&);
 
-        Vector<WebKit::SandboxExtension::Handle> sandboxExtensionHandles;
-        for (auto& element : m_data->elements()) {
-            if (auto* fileData = std::get_if<WebCore::FormDataElement::EncodedFileData>(&element.data)) {
-                const String& path = fileData->filename;
-                if (auto handle = WebKit::SandboxExtension::createHandle(path, WebKit::SandboxExtension::Type::ReadOnly))
-                    sandboxExtensionHandles.append(WTFMove(*handle));
-            }
-        }
-        encoder << WTFMove(sandboxExtensionHandles);
-    }
+    RefPtr<WebCore::FormData> data() const { return m_data.get(); }
+    RefPtr<WebCore::FormData> takeData() { return WTFMove(m_data); }
 
-    static std::optional<FormDataReference> decode(Decoder& decoder)
-    {
-        std::optional<bool> hasFormData;
-        decoder >> hasFormData;
-        if (!hasFormData)
-            return std::nullopt;
-        if (!hasFormData.value())
-            return FormDataReference { };
+    Vector<WebKit::SandboxExtension::Handle> sandboxExtensionHandles() const;
 
-        std::optional<Ref<WebCore::FormData>> formData;
-        decoder >> formData;
-        if (!formData)
-            return std::nullopt;
+private:
+    RefPtr<WebCore::FormData> m_data;
+};
 
-        std::optional<Vector<WebKit::SandboxExtension::Handle>> sandboxExtensionHandles;
-        decoder >> sandboxExtensionHandles;
-        if (!sandboxExtensionHandles)
-            return std::nullopt;
+inline FormDataReference::FormDataReference(RefPtr<WebCore::FormData>&& data, Vector<WebKit::SandboxExtension::Handle>&& sandboxExtensionHandles)
+    : m_data(WTFMove(data))
+{
+    WebKit::SandboxExtension::consumePermanently(sandboxExtensionHandles);
```

Comment:
WTFMove?

Reply:
Wouldn't have any benefit at the moment?
```
bool consumePermanently(const Vector<Handle>&);
```